Update App Store Packaging Guide (#158)
AvaloniaUI/Documentation: docs/distribution-publishing/macos.md

You need a lot of things:

* Your app satisfies [App Store Review Guidelines](https://developer.apple.com/app-store/review/guidelines/).

> * Your app satisfies [MacOS Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/overview/themes/).

Reviewer: this review is a bit pedantic but please use the official "macOS" capitalization on all mentions of it

* Apple Developer Account, with your Apple ID connected to it.
* Your app is registered in [App Store Connect](https://appstoreconnect.apple.com/apps/).
* Transporter app installed from App Store.
* Latest Xcode installed with your Apple ID authorized into it.
* Two certificates: `3rd Party Mac Developer Installer` for signing `.pkg` file and `3rd Party Mac Developer Application` for signing a bundle.
* App Store Provision Profile - get it for your app [here](https://developer.apple.com/account/resources/profiles/list).
* Two entitlements: One for signing `.app` and other for signing app helpers.
* Your app content is [bundled correctly](https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle).
* Your bundle is signed correctly.

> * Your `.dylib` files do not contain non ARM/x64 architectures. You can remove this by using `lipo` command line tool.

Reviewer: ```suggestion
* Your `.dylib` files doesn't contain any non-ARM/x64 architectures. You can remove these by using `lipo` command line tool.
```

> * Your app is ready to be launched inside a [sandbox](https://developer.apple.com/library/archive/documentation/Security/Conceptual/AppSandboxDesignGuide/AboutAppSandbox/AboutAppSandbox.html).

Reviewer: ```suggestion
* Your app is ready to be launched from inside a [sandbox](https://developer.apple.com/library/archive/documentation/Security/Conceptual/AppSandboxDesignGuide/AboutAppSandbox/AboutAppSandbox.html).
```

* Go to Access Control Tab.
* Select `Allow all applications to access this item` in case you don't want to enter a mac profile password for every file sign.

### Sandbox and bundle <a id="sandbox-and-bundle"></a>

App Store required app to be launched inside a sandbox. That means app will have no access to everything and cannot harm user's PC.

Your app should be ready for this and do not crash if any folder is read/write protected.

> .NET 6 is not crushing inside a sandbox only if it's published with single file option enabled. Example:

Reviewer: ```suggestion
.NET 6 apps will not crash inside a sandbox only if it's published with single file option enabled. Example:
```

`dotnet publish src/MyApp.csproj -c Release -f net6.0 -r osx-x64 --self-contained true -p:PublishSingleFile=true`

> Your app content should be bundled correctly. [Here is Apple article with a lot of info](https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle)

Reviewer: ```suggestion
Your app content should be bundled correctly. [Here's an article from Apple with a lot of useful info](https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle).
```

> Most important rules from the article: 

Reviewer: ```suggestion
The most important rules from the article are: 
```

> * `.dll` files are not concidered as a code by Apple. So it should be placed inside `/Resources` folder and can be not signed.

Reviewer: ```suggestion
* `.dll` files are not considered as code by Apple. So it should be placed inside `/Resources` folder and can't signed.
```

> * `/MacOS` files should contain only executable mach-o - you app executable and any other helper executables

Reviewer: ```suggestion
* Inside the `/MacOS` folder, files should contain only executable mach-o - your app's executable and any other helper executables.
```

> * All other mach-o `.dylib` files shoul be inside `Frameworks/` folder.

Reviewer: ```suggestion
* All other mach-o `.dylib` files should be inside the `Frameworks/` folder.
```

> To satisfy this requirement without a lot of pain you can use relative symlinks from `MacOS/` folder to `Resources/` and `Frameworks/` folders. Example command:

Reviewer: ```suggestion
To satisfy this requirement without a lot of pain you can use relative symlinks from `MacOS/` folder to `Resources/` and `Frameworks/` folders. As an example:
```

`ln -s fromFile toFile`

> Also better rewrite resources access code for Mac App Store build to directly access `Resources/` folder without symlinks, because over symlinks you can have IO access issues in sandbox.

Reviewer: ```suggestion
Also it's better to rewrite your app's resources access scheme to directly access `Resources/` folder without using any symlinks, because over symlinks you might get I/O access issues in sandbox.
```

### Sandbox entitlements and signing <a id="sandbox-entitlements-and-signing"></a>

> You should read all [entitlements documentation](https://developer.apple.com/documentation/bundleresources/entitlements) and peek ones your app will access.

Reviewer: ```suggestion
You should read all [entitlements documentation](https://developer.apple.com/documentation/bundleresources/entitlements) and choose the ones your app requires.
```

> First entitlements file is to sign all helper executables inside `.app/Content/MacOS/` folder. It should look like this.

Reviewer: ```suggestion
First for the entitlements file is to sign all helper executables inside `.app/Content/MacOS/` folder. It should look like this.
```

</plist>
```

> Second entitlements file is to sign app executable and a whole app bundle. Should contain all app's permissions. Here is an example:

Reviewer: ```suggestion
Second is to sign app executable and a whole app bundle. It should contain all app's permissions. Here is an example:
```
